Library/brew.sh

```diff
@@ -160,6 +160,16 @@ EOS
   esac
 fi
 
+if [[ "$HOMEBREW_COMMAND" = "install" ]] && [[ -n "$HOMEBREW_DEVELOPER" ]]
+then
+  # Hide shellcheck complaint:
+  # shellcheck source=/dev/null
+  (
+    source "$HOMEBREW_LIBRARY/Homebrew/cmd/update.sh"
+    { homebrew-update --preinstall; }
```

Comment:
I think we should try to not over complicate the logic in `brew.sh` when handling this. Instead, how about create a `cmd/install.sh` and `cmd/install-without-update.rb`. So
* `cmd/install.sh` is like:
```bash
# code to set HOMEBREW_DEBUG and HOMEBREW_VERBOSE based on args.
{ homebrew-update --preinstall; brew install-without-update "$@"; }
```
* `cmd/install-without-update.rb` is the current `cmd/install.rb`.